src/ts/controls/tabs-container.ts

```diff
@@ -63,22 +63,16 @@ export class TabsContainer {
             (item) => this.handleTabFocusEvent(item),
             (x, y, dragListener, item) => this.handleTabDragStartEvent(x, y, dragListener, item));
 
-        if (this._tabs.length === 0) {
-            this._tabs.push(tab);
-            this._element.appendChild(tab.element);
-        } else {
-            if (index === undefined) {
-                index = this._tabs.length;
-            }
-
-            if (index > 0) {
-                this._tabs[index - 1].element.insertAdjacentElement('afterend', tab.element);
-            } else {
-                this._tabs[0].element.insertAdjacentElement('beforebegin', tab.element);
-            }
+        if (index === undefined) {
+            index = this._tabs.length;
+        }
 
-            this._tabs.splice(index, 0, tab);
+        if (index < this._element.childNodes.length) {
+            this._element.insertBefore(tab.element, this._element.childNodes[index]);
+        } else {
+            this._element.appendChild(tab.element);
         }
+        this._tabs.push(tab);
```

Comment:
this looks like it's subtly changing the _tabs array. I think previously _tabs would have the same order as the DOM elements, now the _tabs are decoupled?

Reply:
Fixed in PR #780
Thanks for spotting this